agent/rpc/peering/validate.go

```diff
@@ -36,6 +36,17 @@ func validatePeeringToken(tok *structs.PeeringToken) error {
 		if port < 1 || port > 65535 {
 			return &errPeeringInvalidServerAddress{addr}
 		}
+		return nil
+	}
+	for _, addr := range tok.ManualServerAddresses {
+		if err := validAddr(addr); err != nil {
+			return err
+		}
+	}
+	for _, addr := range tok.ServerAddresses {
+		if err := validAddr(addr); err != nil {
+			return err
+		}
```

Comment:
Should it be an error if both of them are set?

Reply:
Well, right now it sends both of them in the token still. I suppose I could change the logic to not populate those addresses if the manual is specified. I don't know if the tokens are automatically rotated and would have issues, but I can check.